### PR TITLE
feat: add ndarray to uri driver

### PR DESCRIPTION
**Pull request by rutujasurve94**
Links to Jira:
https://jinaai.atlassian.net/browse/JINACORE-215

Adding `NdArray` to `URI` Driver.

### CODE REVIEW

File: jina/drivers/convert.py [hunk 1]
```diff
@@ -134,7 +133,20 @@ def png_pack(png_tag, data):
...
+    
+        arr.uri = 'data:image/png;base64,' + base64.b64encode(png_bytes).decode()
+
+
+class Blob2PngURI(NdArray2PngURI):
+    """Simple DocCrafter used in :command:`jina hello-world`,
+        it reads ``buffer`` into base64 png and stored in ``uri``"""
+
+    def __init__(self, target='uri', width: int = 28, height: int = 28, *args, **kwargs):
+        super().__init__(target,width, height, *args, **kwargs)
```
Comment (maximilianwerk) [inline jina/drivers/convert.py:145]: Just simple formatting issue: a space behind one `,` is missing.

File: jina/drivers/convert.py [hunk 2]
```diff
@@ -101,17 +101,16 @@ def convert(self, d):
...
 
     def __init__(self, target='uri', width: int = 28, height: int = 28, *args, **kwargs):
         super().__init__(target, *args, **kwargs)
         self.width = width
         self.height = height
 
-    def convert(self, d):
-        arr = pb2array(d.blob)
+    def convert(self, arr):
         pixels = []
```
Comment (maximilianwerk) [inline jina/drivers/convert.py:114]: Perhaps put everything into a separate function and only do here
```
arr.uri = self.my_new_function(arr)
```

File: jina/drivers/convert.py [hunk 3]
```diff
@@ -134,7 +133,20 @@ def png_pack(png_tag, data):
...
+class Blob2PngURI(NdArray2PngURI):
+    """Simple DocCrafter used in :command:`jina hello-world`,
+        it reads ``buffer`` into base64 png and stored in ``uri``"""
+
+    def __init__(self, target='uri', width: int = 28, height: int = 28, *args, **kwargs):
+        super().__init__(target,width, height, *args, **kwargs)
+
+    def convert(self, d):
+        arr = pb2array(d.blob)
+        super().convert(arr)
```
Comment (maximilianwerk) [inline jina/drivers/convert.py:149]: And this would then be:
`d.uri = self.my_new_function(arr)

**Review by maximilianwerk — COMMENTED** [review]
I'd love to see a test for this. Not sure how complicated it is, but if you have an idea to implement it fast, please do.

File: jina/drivers/convert.py (hunk 2 above)
Comment (rutujasurve94) [reply jina/drivers/convert.py:113]: done

File: jina/drivers/convert.py (hunk 1 above)
Comment (rutujasurve94) [reply jina/drivers/convert.py]: done

File: jina/drivers/convert.py [hunk 4]
```diff
@@ -134,8 +132,23 @@ def png_pack(png_tag, data):
...
+class Blob2PngURI(NdArray2PngURI):
+    """Simple DocCrafter used in :command:`jina hello-world`,
+        it reads ``buffer`` into base64 png and stored in ``uri``"""
+
+    def __init__(self, target='uri', width: int = 28, height: int = 28, *args, **kwargs):
+        super().__init__(target, width, height, *args, **kwargs)
+
+    def convert(self, d):
+        arr = pb2array(d.blob)
+        d.uri = super().png_convertor(arr)
```
Comment (maximilianwerk) [inline jina/drivers/convert.py:151]: This can now be `self.png_convertor(arr)`. No need for the `super()` magic anymore.

Comment (rutujasurve94) [reply jina/drivers/convert.py]: done

**Review by maximilianwerk — APPROVED** [review]
LGTM

**Pull request merged**